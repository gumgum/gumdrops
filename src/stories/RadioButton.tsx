import React, { ReactChild, ReactChildren } from 'react';
import cx from 'classnames';
import { Colors } from 'types';

export interface RadioButtonProps extends React.InputHTMLAttributes<HTMLInputElement> {
    label?: ReactChild | ReactChildren | Element;
    color?: Colors;
    checked?: boolean;
}

const baseClass = 'gds-form-group';

export const RadioButton: React.FC<RadioButtonProps> = ({
    color = 'default',
    label,
    className,
    ...otherProps
}) => {
    const rootClass = cx(baseClass, `${baseClass}__radio-label`, className, {
        [`${baseClass}--${color}`]: color
    });

    return (
        <label className={rootClass}>
            <input className="gds-form-group__radio-input" {...otherProps} type="radio" />
            <span className="gds-form-group__radio-indicator" />
            {label}
        </label>
    );
};
