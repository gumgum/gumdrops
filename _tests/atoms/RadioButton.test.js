import renderer from 'react-test-renderer';
import React from 'react';
import RadioButton from '../../components/atoms/RadioButton';

const defaultProps = {
    label: 'Cool',
    value: 'cool',
    className: 'custom-class',
    context: 'success',
    name: 'radio-buttons',
    onBlur: jest.fn(),
    onChange: jest.fn()
};

test('Expect <RadioButton> to render properly', () => {
    const tree = renderer.create(<RadioButton {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
});
