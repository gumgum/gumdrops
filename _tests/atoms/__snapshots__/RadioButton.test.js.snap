// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <RadioButton> to render properly 1`] = `
<label
  className="gds-form-group gds-form-group__radio-label custom-class gds-form-group--success"
>
  <input
    className="gds-form-group__radio-input"
    name="radio-buttons"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    type="radio"
    value="cool"
  />
  <span
    className="gds-form-group__radio-indicator"
  />
  Cool
</label>
`;
